### PR TITLE
feat(alert previews): Support issue platform

### LINKED ISSUE #44264: Investigate work needed to add issue platform to rule previews
Andrew's project with alert rule previews https://github.com/getsentry/sentry/blob/master/src/sentry/rules/history/preview.py has a mapping of group categories to datasets https://github.com/getsentry/sentry/blob/master/src/sentry/rules/history/preview_strategy.py#L18-L21 

We should spec out the work needed to add issue platform issues to this feature.

### PR DESCRIPTION
**Pull request by ceorourke**
Add support for issues using the issue platform in alert previews. I refactored some of the dictionaries to be functions instead so there is an easy fallback to the issue platform and less duplication. 

Fixes #44264

### CODE REVIEW

File: src/sentry/rules/history/preview.py [hunk 1]
```diff
@@ -279,7 +276,7 @@ def get_group_dataset(group_ids: Sequence[int]) -> Dict[int, Dataset]:
     """
     group_categories = Group.objects.filter(id__in=group_ids).values_list("id", "type")
     return {
-        group[0]: GROUP_CATEGORY_TO_DATASET.get(get_group_type_by_type_id(group[1]).category)
+        group[0]: get_dataset_from_category(get_group_type_by_type_id(group[1]).category)
```
Comment (barkbarkimashark) [inline src/sentry/rules/history/preview.py:279]: nit: 
```
return {
        group_id: get_dataset_from_category(get_group_type_by_type_id(group_type).category)
        for group_id, group_type in group_categories
    }
```

**Review by barkbarkimashark — APPROVED** [review]
lgtm

**Review by wedamija — COMMENTED** [review]
We might need to have a test for the generic event version of `test_frequency_conditions` as well, since that's making snuba queries.

**Comment by ceorourke** [comment]
I think there's actually a bug here where GroupType and GroupCategory values are mixed up because `GroupCategory.ERROR` and `ErrorGroupType.type_id` are both 1. We have no tests specifically filtering performance issues so it was not caught ... I'll try to figure it out tomorrow.

**Review by wedamija — APPROVED** [review]

**Pull request merged**